libmultilabel/nn/dataset_attentionxml.py

```diff
@@ -0,0 +1,115 @@
+from __future__ import annotations
+
+from typing import Sequence, Optional
+
+import numpy as np
+from numpy import ndarray
+from scipy.sparse import csr_matrix, issparse
+from torch import Tensor, is_tensor
+from torch.utils.data import Dataset
+
+
+class PlainDataset(Dataset):
+    """Plain (compared to nn.data_utils.TextDataset) dataset class for multi-label dataset.
+    WHY EXISTS: The reason why this class is necessary is that it can process labels in sparse format, while TextDataset
+    does not.
+    Moreover, TextDataset implements multilabel binarization in a mandatory way. Nevertheless, AttentionXML already does
+    this while generating clusters. There is no need to do multilabel binarization again.
+
+    Args:
+        x (list | ndarray | Tensor): texts.
+        y (Optional: csr_matrix | ndarray | Tensor): labels.
+    """
+
+    def __init__(self, x, y=None):
+        if y is not None:
+            assert len(x) == y.shape[0], "Sizes mismatch between texts and labels"
+        self.x = x
+        self.y = y
+
+    def __getitem__(self, idx: int) -> tuple[Sequence, ndarray] | tuple[Sequence]:
+        item = {"text": self.x[idx]}
+
+        # train/val/test
+        if self.y is not None:
+            if issparse(self.y):
+                y = self.y[idx].toarray().squeeze(0).astype(np.int32)
+            elif isinstance(self.y, ndarray):
+                y = self.y[idx].astype(np.int32)
+            elif is_tensor(self.y):
+                y = self.y[idx].int()
+            else:
+                raise TypeError(
+                    "The type of y should be one of scipy.csr_matrix, numpy.ndarry, and torch.Tensor."
+                    f"But got {type(self.y)} instead."
+                )
+            item["label"] = y
+        return item
+
+    def __len__(self):
+        return len(self.x)
+
+
+class PLTDataset(PlainDataset):
+    """Dataset for model_1 of AttentionXML.
+
+    Args:
+        x: texts.
+        y: labels.
+        num_classes: number of classes.
+        num_labels_selected: the number of selected labels. Pad any labels that fail to reach this number.
```

Comment:
```suggestion
        num_labels_selected: the number of selected labels.
```
The 2nd sentence is not right. label itself cannot "reach a number." Anyway we don't need to describe the padding here (as it's described later)